openmc/deplete/nuclide.py

```diff
@@ -211,14 +230,243 @@ def to_xml_element(self):
+
+        for g_index, energy in enumerate(energies):
+            prod_map = fission_yields[energy]
+            for prod_ix, product in enumerate(ordered_prod):
+                yield_val = prod_map.get(product)
+                yield_matrix[g_index, prod_ix] = (
+                    0.0 if yield_val is None else yield_val)
+
+        return cls(energies, ordered_prod, yield_matrix)
+
+    def to_xml_element(self, root):
+        """Write fission yield data to an xml element
+
+        Parameters
+        ----------
+        root : xml.etree.ElementTree.Element
+            Element to write distribution data to
+        """
+        for energy, yield_obj in self.items():
+            yield_element = ET.SubElement(root, "fission_yields")
+            yield_element.set("energy", str(energy))
+            product_elem = ET.SubElement(yield_element, "products")
+            product_elem.text = " ".join(map(str, yield_obj.products))
+            data_elem = ET.SubElement(yield_element, "data")
+            data_elem.text = " ".join(map(str, yield_obj.yields))
+
+
+class FissionYield(Mapping):
+    """Mapping for fission yields of a parent at a specific energy
+
+    Separated to support nested dictionary-like behavior for
+    :class:`FissionYieldDistribution`, and allowing math operations
+    on a single vector of yields. Can in turn be used like a
+    dictionary to fetch fission yields.
+    Supports multiplication of a scalar to scale the fission
+    yields and addition of another set of yields.
+
+    Does not support resizing / inserting new products that do
+    not exist.
+
+    Parameters
+    ----------
+    products : tuple of str
+        Products for this specific distribution
+    yields : numpy.ndarray
+        Fission product yields for each product in ``products``
+
+    Attributes
+    ----------
+    products : tuple of str
+        Products for this specific distribution
+    yields : numpy.ndarray
+        Fission product yields for each product in ``products``
+
+    Examples
+    --------
+    >>> import numpy
+    >>> fy_vector = FissionYield(
+    ...     ("Xe135", "I129", "Sm149"),
+    ...     numpy.array((0.002, 0.001, 0.0003)))
+    >>> fy_vector["Xe135"]
+    0.002
+    >>> new = fy_vector.copy()
+    >>> fy_vector *= 2
+    >>> fy_vector["Xe135"]
+    0.004
+    >>> new["Xe135"]
+    0.002
+    >>> (new + fy_vector)["Sm149"]
+    0.0009
+    >>> dict(new)
+    {"Xe135": 0.002, "I129": 0.001, "Sm149": 0.0003}
+    """
+
+    def __init__(self, products, yields):
+        self.products = products
+        self.yields = yields
+
+    def __getitem__(self, product):
+        if product not in self.products:
+            raise KeyError(product)
+        return self.yields[self.products.index(product)]
+
+    def __len__(self):
+        return len(self.products)
+
+    def __iter__(self):
+        return iter(self.products)
+
+    def items(self):
+        return zip(self.products, self.yields)
```

Comment:
Correct, which is awesome. I added this because we can improve the iteration because `products` and `yields` are paired iterables. The mixin version of `items` is essentially
```
for k in self:
   return k, self[k]
```

By zipping like this, we avoid costly calls to `__getitem__` due to the `index` method and iterate a lot faster.  Now that I think about it, we could probably overwrite the `keys` and `values` mixin methods as well to take advantage of the structure of `products` and `yields`

Reply:
Ah, good point